src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs

```diff
@@ -236,6 +242,9 @@ private async Task DoSend()
 
                 // Complete the output after disposing the socket
                 Output.Complete(error);
+
+                // Cancel any pending flushes so that the input loop is un-paused
+                Input.CancelPendingFlush();
```

Comment:
Either way it seems like a good idea to clean up the transport side of things as quickly as possible, but was not doing this before causing any major issues in practice? There's a lot of cleanup that's going to be delayed anyway until the app starts trying to consume the input and/or observes the connection closed.

Reply:
Yes it was causing a test to fail. I'll find that test. Libuv works fine, sockets doesn't.